src/uwtools/drivers/sfc_climo_gen.py

```diff
@@ -0,0 +1,162 @@
+"""
+A driver for sfc_climo_gen.
+"""
+
+import os
+import stat
+from pathlib import Path
+from typing import Any, Dict
+
+from iotaa import asset, dryrun, task, tasks
+
+from uwtools.config.formats.nml import NMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.utils.processing import execute
+from uwtools.utils.tasks import file
+
+
+class SfcClimoGen(Driver):
+    """
+    A driver for sfc_climo_gen.
+    """
+
+    def __init__(self, config_file: Path, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config_file: Path to config file.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config_file=config_file, dry_run=dry_run, batch=batch)
+        if self._dry_run:
+            dryrun()
+        self._rundir = Path(self._driver_config["run_dir"])
+
+    # Workflow tasks
+
+    @task
+    def namelist_file(self):
+        """
+        The namelist file.
+        """
+        fn = "fort.41"
+        yield self._taskname(f"namelist file {fn}")
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        vals = self._driver_config["namelist"]["update_values"]["config"]
+        input_paths = [Path(v) for k, v in vals.items() if k.startswith("input_")]
+        input_paths += [Path(vals["mosaic_file_mdl"])]
+        input_paths += [Path(vals["orog_dir_mdl"]) / fn for fn in vals["orog_files_mdl"]]
+        yield [file(input_path) for input_path in input_paths]
```

Comment:
In the current implementation, the `namelist_file` task collects the names of all the input files needed by `sfc_climo_gen`, as specified in its `config` namelist, and yields them as `@external` tasks, to ensure that the namelist file is not created if any of the files it specifies do not exist. This could potentially be done instead in the `_run_via_batch_submission()` method, so that `uw` would decline to submit a batch job that is (probably?) doomed to fail, which would be a waste of time and other resources. But doing the check here essentially tells the user "Your namelist is incorrect", which might be useful.

Reply:
We saw it just now in the live demo but, for `@external` tasks like the ones representing the files names as inputs in the namelist, `iotaa` will emit `WARNING` messages about those, e.g.
```
[2024-02-21T18:58:45]  WARNING File /scratch1/NCEPDEV/nems/role.epic/UFS_SRW_data/develop/fix/fix_sfc_climo/facsf.1.0.ncx: State: Pending (EXTERNAL)
```

in the case of a file that doesn't exist (I changed `.nc` to `.ncx` in the filename in the YAML config to make this break) vs e.g.
```
[2024-02-21T18:58:45]     INFO File /scratch1/NCEPDEV/nems/role.epic/UFS_SRW_data/develop/fix/fix_sfc_climo/maximum_snow_albedo.0.05.nc: State: Ready
```

for an `@external` file that exists. These `WARNING` message stand out well in the `uw` output IMO, and users will quickly get used to paying attention to them.